core/lib/__tests__/richText.test.ts

```diff
@@ -0,0 +1,199 @@
+			pubId,
+			elements,
+			newValues,
+		});
+		expect(values).toEqual(newValues);
+	});
+
+	it("should not overwrite if another field is not referenced", () => {
+		// getPubValues returns an empty dictionary if we only have text (no pubfields, pubs, etc.)
+		vi.mocked(getPubValues).mockReturnValue({});
+
+		const elements = [
+			{ slug: "croccroc:title", schemaName: CoreSchemaType.String },
+			{ slug: "croccroc:richtext", schemaName: CoreSchemaType.RichText },
+		];
+		const { values } = parseRichTextForPubFieldsAndRelatedPubs({
+			pubId,
+			elements,
+			newValues: {
+				"croccroc:title": "original title",
+				"croccroc:richtext": "todo",
+			},
+		});
+		expect(values).toEqual({
+			"croccroc:title": "original title",
+			"croccroc:richtext": "todo",
+		});
+	});
+
+	it("should overwrite pubfields", () => {
+		// Adding a Title field
+		vi.mocked(getPubValues).mockReturnValue({
+			"": {
+				pubId: "",
+				parentPubId: "",
+				pubTypeId: "6a944264-3c0a-47e1-8589-c5bed4448f35",
+				values: {
+					"croccroc:title": [
+						{
+							type: "paragraph",
+							attrs: {
+								id: null,
+								class: null,
+							},
+							content: [
+								{
+									type: "text",
+									text: "new title",
+								},
+							],
+						},
+					],
+				},
+			},
+		});
+		const elements = [
+			{ slug: "croccroc:title", schemaName: CoreSchemaType.String },
+			{ slug: "croccroc:richtext", schemaName: CoreSchemaType.RichText },
+		];
+		const { values } = parseRichTextForPubFieldsAndRelatedPubs({
+			pubId,
+			elements,
+			newValues: {
+				"croccroc:richtext": "todo",
+			},
+		});
+		expect(values).toEqual({
+			"croccroc:richtext": "todo",
+			"croccroc:title": "new title",
+		});
+	});
+
+	it("should collapse multipart fields", () => {
+		// Adding a Title field with two parts
+		vi.mocked(getPubValues).mockReturnValue({
+			"": {
+				pubId: "",
+				parentPubId: "",
+				pubTypeId: "6a944264-3c0a-47e1-8589-c5bed4448f35",
+				values: {
+					"croccroc:title": [
+						{
+							type: "paragraph",
+							attrs: {
+								id: null,
+								class: null,
+							},
+							content: [
+								{
+									type: "text",
+									text: "new title",
+								},
+							],
+						},
+						{
+							type: "paragraph",
+							attrs: {
+								id: null,
+								class: null,
+							},
+							content: [
+								{
+									type: "text",
+									text: "second part",
+								},
+							],
+						},
+					],
+				},
+			},
+		});
+		const elements = [
+			{ slug: "croccroc:title", schemaName: CoreSchemaType.String },
+			{ slug: "croccroc:richtext", schemaName: CoreSchemaType.RichText },
+		];
+		const { values } = parseRichTextForPubFieldsAndRelatedPubs({
+			pubId,
+			elements,
+			newValues: {
+				"croccroc:richtext": "todo",
+			},
+		});
+		expect(values).toEqual({
+			"croccroc:richtext": "todo",
+			"croccroc:title": "new title, second part",
+		});
+	});
+
+	it("returns children pub", () => {
```

Comment:
this isn't used yet, but I _think_ is how adding related pubs will work. again, this is kind of dependent on the "main content" issue